From c0c014f7409a9dd2d3a4d340f91cd1fab8b9846a Mon Sep 17 00:00:00 2001
From: Leo Yan <leo.yan@linaro.org>
Date: Thu, 16 Jun 2016 14:56:54 +0800
Subject: [PATCH 1/1] thermal: hisilicon: use dev_dbg when bind sensors

Current code use dev_err to output log if kernel cannot bind any one
sensor in initialization. This is a strictly checking to make sure all
four sensors have to be used for system.

After enable thermal power allocator, usually system only use one sensor
for multiple actors. This patch changes to use dev_dbg so that avoid
confusion in boot log, and remove duplicate printing log.

Signed-off-by: Leo Yan <leo.yan@linaro.org>
---
 drivers/thermal/hisi_thermal.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/thermal/hisi_thermal.c b/drivers/thermal/hisi_thermal.c
index f642966..06baf72 100644
--- a/drivers/thermal/hisi_thermal.c
+++ b/drivers/thermal/hisi_thermal.c
@@ -260,7 +260,7 @@ static int hisi_thermal_register_sensor(struct platform_device *pdev,
 	if (IS_ERR(sensor->tzd)) {
 		ret = PTR_ERR(sensor->tzd);
 		sensor->tzd = NULL;
-		dev_err(&pdev->dev, "failed to register sensor id %d: %d\n",
+		dev_dbg(&pdev->dev, "failed to register sensor id %d: %d\n",
 			sensor->id, ret);
 		return ret;
 	}
@@ -351,10 +351,7 @@ static int hisi_thermal_probe(struct platform_device *pdev)
 	for (i = 0; i < HISI_MAX_SENSORS; ++i) {
 		ret = hisi_thermal_register_sensor(pdev, data,
 						   &data->sensors[i], i);
-		if (ret)
-			dev_err(&pdev->dev,
-				"failed to register thermal sensor: %d\n", ret);
-		else
+		if (!ret)
 			hisi_thermal_toggle_sensor(&data->sensors[i], true);
 	}
 
-- 
2.10.2

